src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -208,12 +209,24 @@ private async Task InnerOnConnectionAsync(ConnectionContext context)
                     await sslStream.DisposeAsync();
                     return;
                 }
-                catch (Exception ex) when (ex is IOException || ex is AuthenticationException)
+                catch (IOException ex)
                 {
                     _logger?.LogDebug(1, ex, CoreStrings.AuthenticationFailed);
                     await sslStream.DisposeAsync();
                     return;
                 }
+                catch (AuthenticationException ex)
+                {
+                    _logger?.LogDebug(1, ex, CoreStrings.AuthenticationFailed);
+                    if (_serverCertificate != null &&
+                        CertificateManager.IsHttpsDevelopmentCertificate(_serverCertificate) &&
+                        !CertificateManager.CheckDeveloperCertificateKey(_serverCertificate))
```

Comment:
There are likely other ways to cause an AuthenticationException besides a bad dev cert, so we do want to constrain it a bit.

Reply:
I started in that point, the problem is that I felt we would be spamming people for example when there wasn't a suitable cipher suite, and this guidance would be useless in that case, and I think that's a common enough problem to avoid placing blame on the cert without issue.

What I'm trying to say is that I would be hesitant to provide blanked guidance on blaming the dev-cert for all AuthenticationExceptions as I think we would get misleading bugs in the form, my dev-cert doesn't work and I'm running these two things and doesn't fix it.

Does it make sense?

Reply:
That makes sense! Thanks!